base_revision/models/base_revision.py

```diff
@@ -67,16 +81,20 @@ def _get_new_rev_data(self, new_rev_number):
             "old_revision_ids": [(4, self.id, False)],
         }
 
+    def _prepare_revision_data(self, new_revision):
+        vals = {"active": False, "current_revision_id": new_revision.id}
+        if "state" in self._fields:
+            vals.update({"state": "cancel"})
```

Comment:
Why assume that the inherited model is going to have such selection value in `state`? Do this in the inherited model. That's why we have put the hook.